## EJERCICIO 4

Abro el dataset poniendo el numero de canción como index.

In [19]:
import pandas as pd
df = pd.read_csv('data/train.csv').set_index('Number')
df

title                artist           top genre  \
Number                                                                        
175          She Looks So Perfect   5 Seconds of Summer            boy band   
577                    Run To You           Bryan Adams          album rock   
275                  Stuck On You         Lionel Richie     adult standards   
514       The Rhythm of the Night                Corona     bubblegum dance   
300               Beautiful Girls         Sean Kingston           dance pop   
...                           ...                   ...                 ...   
162     Guilty (feat. Barry Gibb)      Barbra Streisand     adult standards   
558            Please Mr. Postman       The Marvelettes  brill building pop   
568     Hypnotize - 2007 Remaster  The Notorious B.I.G.  east coast hip hop   
292               The Way We Were      Barbra Streisand     adult standards   
126                    In the End           Linkin Park   alternative metal   

        year  bpm  nrgy  dnce  dB  live  val  dur  acous  spch  pop  
Number                                                               
175     2014  160    95    49  -4    33   44  202      0    13   69  
577     1984  127    81    56  -6    29   64  233      1     3   65  
275     1983  132    32    51 -16     9   25  192     62     3   69  
514     1994  128    81    69 -12    10   80  264      0     3   69  
300     2014  118    64    54  -7    24   75  242     14     9   60  
...      ...  ...   ...   ...  ..   ...  ...  ...    ...   ...  ...  
162     1980  151    43    63 -16    11   95  264     23     4   67  
558     1961  122    78    79  -5    18   97  148     73     3   68  
568     2007   94    69    90  -4    29   63  230     18    28   68  
292     1974  138    32    26 -12    13   17  211     89     3   60  
126     2000  105    86    56  -6    21   40  217      1     6   83  

[386 rows x 14 columns]

De estos 386 registros, tenemos que sacar cuantas canciones por artista hay:

In [31]:
df_artist = df[['artist']].copy()
conteo = pd.DataFrame(df_artist.groupby(['artist'])['artist'].count().sort_values(ascending=False))
conteo.head(10)


artist
artist                  
Elton John             9
ABBA                   8
Barbra Streisand       5
The Beatles            5
Queen                  4
Mariah Carey           4
Rihanna                4
Taio Cruz              4
Whitney Houston        4
Ariana Grande          3

Saco como query SQL:

In [34]:
SELECT
    artist AS ARTISTA,
    COUNT(artist) AS CONTEO
FROM
    spotify_dataset
GROUP BY artist
SORT BY 2 DESC

SyntaxError: invalid syntax (1057038556.py, line 1)